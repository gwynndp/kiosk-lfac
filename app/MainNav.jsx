import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

const Nav = styled.div`
  background-color: blue;
  width: 130px;
  padding: 20px 0px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

const ButtonLink = styled(Link)`
  background: none;
  border: none;
  color: white;
  text-transform: uppercase;
  padding: 15px;
  display: flex;
  flex-wrap: wrap;
`;

const Button = styled.button`
  background: none;
  border: none;
  color: white;
  text-transform: uppercase;
  padding: 15px;
  display: flex;
  flex-wrap: wrap;
`;

function handleScrollNext(e) {
  const target = document.querySelector("#gallery");
  // const height = target.offsetHeight;  //height of the scroll space
  // if target.scrollTop === target.scrollTopMax then it is at the bottom of the scrolling content
  // if target.scrollTop === 0 then it is at the top of the scrolling content
  e.preventDefault();
  const adjustment = e.target.parentNode.id === "next" ? 365 : -365;
  target.scrollBy({
    top: adjustment,
    behavior: "smooth"
  });
}

export default function MainNav({ gallery }) {
  console.log("INSIDE MAIN NAV", gallery);
  return (
    <Nav>
      <ButtonLink exact="true" to="/">
        <img src="images/kiosk_home.png" alt="home" width="100px" />
      </ButtonLink>
      {gallery === "true" ? (
        <>
          <Button id="prev" onClick={e => handleScrollNext(e)}>
            <div>PREV</div>
            <img
              src="images/kiosk_arrow_up.png"
              alt="scroll up"
              width="100px"
            />
          </Button>
          <Button id="next" onClick={e => handleScrollNext(e)}>
            <img
              src="images/kiosk_arrow_down.png"
              alt="scroll down"
              width="100px"
            />
            <div>NEXT</div>
          </Button>
        </>
      ) : null}
    </Nav>
  );
}

MainNav.propTypes = {
  gallery: PropTypes.string.isRequired
};
