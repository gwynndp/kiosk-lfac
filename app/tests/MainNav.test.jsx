import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import MainNav from "../MainNav";

test("should render a MainNav", () => {
  const tree = renderer
    .create(
      <Router>
        <MainNav gallery="true" />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// should show home button on every page
// should show the prev and next buttons on the gallery pages

// should display on every page?
