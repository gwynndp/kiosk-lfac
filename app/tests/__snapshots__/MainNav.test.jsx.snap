// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a MainNav 1`] = `
<div
  className="sc-AykKC duquAX"
>
  <a
    className="sc-AykKD dWbVqE"
    exact="true"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="home"
      src="images/kiosk_home.png"
      width="100px"
    />
  </a>
  <button
    className="sc-AykKE gbAOGp"
    id="prev"
    onClick={[Function]}
  >
    <div>
      PREV
    </div>
    <img
      alt="scroll up"
      src="images/kiosk_arrow_up.png"
      width="100px"
    />
  </button>
  <button
    className="sc-AykKE gbAOGp"
    id="next"
    onClick={[Function]}
  >
    <img
      alt="scroll down"
      src="images/kiosk_arrow_down.png"
      width="100px"
    />
    <div>
      NEXT
    </div>
  </button>
</div>
`;
